src/AdoNetCore.AseClient/AseDecimal.cs

```diff
@@ -0,0 +1,564 @@
+        /// <returns>An AseDecimal structure with the specified precision and scale.</returns>
+        public AseDecimal ToAseDecimal(int outputPrecision, int outputScale)
+        {
+            if (outputPrecision == Precision && outputScale == Scale)
+            {
+                return new AseDecimal(new BigDecimal(Backing.Mantissa, Backing.Exponent));
+            }
+
+            var mantissa = Backing.Mantissa;
+
+            if (outputPrecision != Precision)
+            {
+                var precisionDifference = Precision - outputPrecision;
+
+                if (precisionDifference < 0)
+                {
+                    mantissa *= BigInteger.Pow(10, -precisionDifference);
+                }
+                else if (precisionDifference > 0)
+                {
+                    mantissa /= BigInteger.Pow(10, precisionDifference);
+                }
+            }
+
+            if (outputScale != Scale)
+            {
+                var scaleDifference = Scale - outputScale;
+
+                if (scaleDifference < 0)
+                {
+                    mantissa /= BigInteger.Pow(10, -scaleDifference);
+                }
+            }
+
+            return Truncate(new AseDecimal(new BigDecimal(mantissa, Backing.Exponent)), outputScale);
+        }
+
+        /// <summary>
+        /// Returns a string representation of this AseDecimal.
+        /// </summary>
+        /// <returns>A string representation of this AseDecimal.</returns>
+        public override string ToString()
+        {
+            return Backing.ToString();
+        }
+
+        public sbyte ToSByte()
+        {
+            return (sbyte)this;
+        }
+
+        public byte ToByte()
+        {
+            return (byte)this;
+        }
+
+        public short ToInt16()
+        {
+            return (short)this;
+        }
+
+        public ushort ToUInt16()
+        {
+            return (ushort)this;
+        }
+
+        public uint ToUInt32()
+        {
+            return (uint)this;
+        }
+
+        public ulong ToUInt64()
+        {
+            return (ulong)this;
+        }
+
+        public int ToInt32()
+        {
+            return (int)this;
+        }
+
+        public long ToInt64()
+        {
+            return (long)this;
+        }
+
+        public float ToSingle()
+        {
+            return (float)this;
+        }
+
+        public double ToDouble()
+        {
+            return (double)this;
+        }
+
+        public decimal ToDecimal()
+        {
+            return (decimal)this;
+        }
+
+        public static AseDecimal Floor(AseDecimal n)
+        {
+            return new AseDecimal(n.Backing.Floor());
+        }
+
+        public static AseDecimal Truncate(AseDecimal n, int outputScale)
+        {
+            if (outputScale < 0)
+            {
+                Logger.Instance?.WriteLine($"Invalid outputScale {outputScale}");
+                throw new AseException("Invalid value.", 30037);
```

Comment:
@senseibaka - is this the same error code raised by the SAP version?

Reply:
Yep